Neos.Flow/Classes/Package.php

```diff
@@ -109,9 +111,9 @@ public function boot(Core\Bootstrap $bootstrap)
 
         $dispatcher->connect(Command\CoreCommandController::class, 'finishedCompilationRun', Security\Authorization\Privilege\Method\MethodPrivilegePointcutFilter::class, 'savePolicyCache');
 
-        $dispatcher->connect(Security\Authentication\AuthenticationProviderManager::class, 'authenticatedToken', function () use ($bootstrap) {
+        $dispatcher->connect(Security\Authentication\AuthenticationProviderManager::class, 'authenticatedToken', function (TokenInterface $token) use ($bootstrap) {
             $session = $bootstrap->getObjectManager()->get(Session\SessionInterface::class);
-            if ($session->isStarted()) {
+            if ($session->isStarted() && ! $token instanceof SessionlessTokenInterface) {
```

Comment:
```suggestion
            if ($session->isStarted() && !$token instanceof SessionlessTokenInterface) {
```

Reply:
Are you sure, about that? What - after applying binding strengths of the operators there is:
`&& (!($token instanceof SessionlessTokenInterface))`. So for me at least if feels strange to bring it next to token, if the relation to token is less strong, than the relation to `&&` ;)

Reply:
I know what you mean, it's unobvious that this is a "not instanceof" instead of a nonesense condition. But we have that style in all other places where we have such a check. So that would rather need a generic code style decision.

Reply:
Fine for me. Didn't notice until now.